caikit_nlp/modules/token_classification/transformer_sentence_classification.py

```diff
@@ -0,0 +1,218 @@
+                (Optional) Labels to output
+            labels_mapping: Dict[str, str]
+                (Optional) Mapping of model labels to more semantically meaningful labels
+        """
+        super().__init__()
+        error.type_check("<NLP12578168E>", str, lang=lang)
+        error.type_check(
+            "<NLP79642537E>", SentenceSplitBase, sentence_splitter=sentence_splitter
+        )
+        error.type_check(
+            "<NLP35742128E>",
+            SequenceClassification,
+            sequence_classifier=sequence_classifier,
+        )
+        error.type_check("<NLP63802045E>", float, default_threshold=default_threshold)
+        error.type_check_all(
+            "<NLP71653678E>", str, allow_none=True, labels_to_output=labels_to_output
+        )
+        error.type_check(
+            "<NLP56932573E>", Dict, allow_none=True, labels_mapping=labels_mapping
+        )
+        self.lang = lang
+        self.sentence_splitter = sentence_splitter
+        self.sequence_classifier = sequence_classifier
+        self.default_threshold = default_threshold
+        self.labels_to_output = labels_to_output
+        self.labels_mapping = labels_mapping
+
+    ################################## API functions #############################################
+
+    def run(
+        self, text: str, threshold: Optional[float] = None
+    ) -> TokenClassificationResult:
+        """Run sentence-level classification on text. Returns results
+        based on score threshold for labels that are to be outputted
+
+        Args:
+            text: str
+                Document to run sentence-level classification on
+            threshold: float
+                (Optional) Threshold based on which to return score results
+
+        Returns:
+            TokenClassificationResult
+        """
+        if threshold is None:
+            threshold = self.default_threshold
+        token_classification_results = []
+        # Split document into sentences
+        sentence_span_list = self.sentence_splitter.run(text)
+        # Run each sentence through the classifier and determine based
+        # on threshold and labels_to_output what results should be returned
+        text_list = [span.text for span in sentence_span_list]
+        classification_results = self.sequence_classifier.run_batch(text_list)
+        for idx, classification_result in enumerate(classification_results):
+            # Each classification prediction is list of classifications
+            # for that particular text example
+            for classification in classification_result.results:
+                # Map labels to semantic labels if provided
+
+                # NOTE: labels need to be specified as str for config
+                label = str(classification.label)
+                if self.labels_mapping:
+                    # Use original classifier label if not found
+                    label = self.labels_mapping.get(label, label)
```

Comment:
why do we want to map the labels to something else and where is a user / model getting the labels_mapping from? It seems if the model has some kind of mapping, shouldn't it be the labels themselves? Seems a bit of convoluted concepts, since if I train a model with X label set, I would want the model to return X labelset, so am confused about this mapping concept.

Reply:
This was done with some semantic mapping in mind, i.e. say a base model just returns 0s and 1s. If a user wants to use this module to essentially as a "detector" to look for the presence of a particular label, 0's and 1's may not mean much. If mapping information can be stored with the model (through config) e.g. 0->`no_X` vs. 1->`has_X` this can help with usability

Reply:
Discussed offline - this will be removed for now. If still needed this should go one level down to `sequence_classification` since the labels are associated with the classifier